python/cudf/cudf/core/reshape.py

```diff
@@ -278,9 +300,27 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
                 result = objs[0].copy()
             return result
         else:
-            return cudf.DataFrame._concat(
-                objs, axis=axis, ignore_index=ignore_index, sort=sort
+            result = cudf.DataFrame._concat(
+                objs,
+                axis=axis,
+                join=join,
+                ignore_index=ignore_index,
+                sort=sort,
             )
+            if ignore_index and join == "inner" and result.empty:
+                indexes = []
+                for obj in objs:
+                    indexes.append(len(obj.index))
+                    result.index = cudf.RangeIndex(sum(indexes))
```

Comment:
Should setting `result.index` be outside of the for loop? Also we can simplify this similar to above:
```suggestion
                result.index = cudf.RangeIndex(sum(len(obj) for obj in objs))
```

Reply:
Also, since we're already calling `cudf.DataFrame._concat` above and passing in `ignore_index` there, wouldn't the resulting index already be a `RangeIndex`?

Reply:
Yes, I think that's correct, but it returns an empty index list, whereas Pandas returns the above. I'm not sure if there's another way to change the index, that's a better route to take than this?

Reply:
I've also added a few changes to try and deal with the differences when we don't have overlapping columns. It's passing all the added tests now :D